### PR TITLE
Scratch images

### PR DESCRIPTION
**Pull request by jonnydawg**
Changes directory names to `hyphenated-lowercase` and updates the Dockerfiles accordingly.

Also moves check images to `golang:1.14.0` and to build from `scratch`

### CODE REVIEW

**Comment by jonnydawg** [comment]
Addresses #349 and #354

**Comment by jonnydawg** [comment]
Haven't touched the `README`s or the the `registry`

**Comment by jonnydawg** [comment]
I think that's everything? Please double check to make sure

File: cmd/check-reaper/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,10 @@
-FROM golang:1.13-alpine AS builder
+FROM golang:1.14.0 AS builder
```
Comment (integrii) [inline cmd/check-reaper/Dockerfile]:
```suggestion
FROM golang AS builder
```

Comment (integrii) [reply cmd/check-reaper/Dockerfile]: let's build with the latest golang... this ensures the newest fixes and all

File: cmd/daemon-set-check/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM golang:1.14.0 AS builder
```
Comment (integrii) [inline cmd/daemon-set-check/Dockerfile]:
```suggestion
FROM golang AS builder
```

File: cmd/deployment-check/Dockerfile [hunk 3]
```diff
@@ -1,8 +1,8 @@
-FROM golang:1.13 AS builder
+FROM golang:1.14.0 AS builder
```
Comment (integrii) [inline cmd/deployment-check/Dockerfile]:
```suggestion
FROM golang AS builder
```

File: cmd/dns-status-check/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
+FROM golang:1.14.0 AS builder
```
Comment (integrii) [inline cmd/dns-status-check/Dockerfile]:
```suggestion
FROM golang AS builder
```

File: cmd/http-check/Dockerfile [hunk 5]
```diff
@@ -1,8 +1,8 @@
-FROM golang:1.13 AS builder
+FROM golang:1.14.0 AS builder
```
Comment (integrii) [inline cmd/http-check/Dockerfile]:
```suggestion
FROM golang AS builder
```

File: cmd/http-content-check/Dockerfile [hunk 6]
```diff
@@ -1,4 +1,4 @@
-FROM golang:alpine
+FROM golang:1.14.0
```
Comment (integrii) [inline cmd/http-content-check/Dockerfile]:
```suggestion
FROM golang
```

File: cmd/kiam-check/Dockerfile [hunk 7]
```diff
@@ -1,8 +1,8 @@
-FROM golang:1.13 AS builder
+FROM golang:1.14.0 AS builder
```
Comment (integrii) [inline cmd/kiam-check/Dockerfile]:
```suggestion
FROM golang AS builder
```

File: cmd/pod-restarts-check/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,10 @@
+FROM golang:1.14.0 AS builder
```
Comment (integrii) [inline cmd/pod-restarts-check/Dockerfile]:
```suggestion
FROM golang AS builder
```

File: cmd/pod-status-check/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,8 @@
+FROM golang:1.14.0 AS builder
```
Comment (integrii) [inline cmd/pod-status-check/Dockerfile]:
```suggestion
FROM golang AS builder
```

File: cmd/test-external-check/Dockerfile [hunk 10]
```diff
@@ -1,9 +1,9 @@
-FROM golang:1.13 AS builder
+FROM golang:1.14.0 AS builder
```
Comment (integrii) [inline cmd/test-external-check/Dockerfile]:
```suggestion
FROM golang AS builder
```

File: cmd/test-external-check/Dockerfile [hunk 11]
```diff
@@ -1,9 +1,9 @@
...
 ADD . /build
 WORKDIR /build/cmd/test-external-check
 RUN go build -v
 
-FROM golang:1.13
+FROM scrapinghub/scrapinghub-stack-hworker
```
Comment (integrii) [inline cmd/test-external-check/Dockerfile]:
```suggestion
FROM scratch
```

File: cmd/test-external-check/Dockerfile [hunk 12]
```diff
@@ -1,9 +1,9 @@
-FROM golang:1.13 AS builder
+FROM golang:1.14.0 AS builder
 ADD . /build
 WORKDIR /build/cmd/test-external-check
 RUN go build -v
```
Comment (integrii) [inline cmd/test-external-check/Dockerfile]:
```suggestion
RUN CGO_ENABLED=0 go build -v
```

File: cmd/check-reaper/Dockerfile (hunk 1 above)
Comment (jonnydawg) [reply cmd/check-reaper/Dockerfile]: isn't it bad practice to build with `latest`?

**Pull request merged**